servers/credentialagent/src/entities/presentations/orchestrators/share-presentation.js

```diff
@@ -81,7 +81,7 @@ const sharePresentation = async (presentation, context) => {
 
   const user = vendorUser
     ? await repos.vendorUserIdMappings.findOrInsertVendorUser(vendorUser)
-    : undefined;
+    : await repos.vendorUserIdMappings.addAnonymousUser();
```

Comment:
if this is shareIdentification, i dont think we want to create the anonymous user. Is this achieving that requirement

Reply:
> if this is shareIdentification, i dont think we want to create the anonymous user. Is this achieving that requirement

The requirement of a vendor/identified user in the case of identification (issuing exchange), but an anonymous user in the case of inspection (disclosure exchange)? I think so.

We have here:

```
const { checkedCredentials, vendorUser } = await doSharePresentation({...}, context);
const user = vendorUser
  ? await repos.vendorUserIdMappings.findOrInsertVendorUser(vendorUser)
  : await repos.vendorUserIdMappings.addAnonymousUser();
```

That `vendorUser`  returned from `doSharePresentation(...)` is defined when internally the `shareIdentification` is called, and is not defined when internally the disclosure exchange related codepaths are called.